src/makeCalculator.js

```diff
@@ -4,7 +4,31 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    add: function(number) {
+      return this.result + number;
+    },
+    subtract: function(number) {
+      return this.result - number;
+    },
+    multiply: function(number) {
+      return this.result * number;
+    },
+    divide: function(number) {
+      return this.result / number;
+    },
+    reset: function() {
+      this.result = 0;
+
+      return this;
+    },
+    operate: function(callback, number) {
+      this.result = callback.call(this, number);
+
+      return this;
+    },
+    result: 0,
```

Comment:
The function `makeCalculator` should return an object named `calculator` to adhere to the naming conventions as per the checklist.